services/federator/src/Federator/Response.hs

```diff
@@ -168,14 +172,24 @@ runFederator env =
           DiscoveryFailure
         ]
     . runInputConst env
-    . runInputSem (embed @IO (readIORef (view domainConfigs env)))
+    . runInputSem (embed @IO (getFederationDomainConfigs env))
     . runInputSem (embed @IO (readIORef (view http2Manager env)))
     . runInputConst (view runSettings env)
     . interpretServiceHTTP
     . runDNSLookupWithResolver (view dnsResolver env)
     . runFederatorDiscovery
     . interpretRemote
 
+getFederationDomainConfigs :: Env -> IO FedUp.FederationDomainConfigs
+getFederationDomainConfigs env = do
+  let mgr = env ^. httpManager
+      Endpoint h p = env ^. service $ Brig
+      baseurl = BaseUrl Http (cs h) (fromIntegral p) ""
+      clientEnv = mkClientEnv mgr baseurl
+  FedUp.getFederationDomainConfigs clientEnv >>= \case
+    Right v -> pure v
+    Left e -> error $ show e
```

Comment:
i guess there is a way in polysemy to throw this through an error effect?

Reply:
@fisx Something like this would work. The exact error value could be different, I extended `ServerError` with this new value.

```
getFederationDomainConfigs ::
  ( Member (Embed IO) r
  , Member (Error ServerError) r
  ) =>
  Env ->
  Sem r FedUp.FederationDomainConfigs
getFederationDomainConfigs env = do
  let mgr = env ^. httpManager
      Endpoint h p = env ^. service $ Brig
      baseurl = BaseUrl Http (cs h) (fromIntegral p) ""
      clientEnv = mkClientEnv mgr baseurl
  liftIO (FedUp.getFederationDomainConfigs clientEnv) >>= \case
    Right v -> pure v
    Left e -> throw . InternalError . pack $ show e
   ```

Reply:
> 

If in doubt, shove it into config. Maybe a total retry time?